tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -836,17 +835,125 @@ export class WebGPUBackend extends KernelBackend {
     return output;
   }
 
+  private getFromPixelTextureLayout(useImport: boolean): WebGPULayout {
+    if (useImport) {
+      if (this.fromPixelImportTextureLayout === null) {
+        this.fromPixelImportTextureLayout =
+            this.createFromPixelTextureLayout(true);
+      }
+      return this.fromPixelImportTextureLayout;
+    }
+
+    if (this.fromPixelTextureLayout === null) {
+      this.fromPixelTextureLayout = this.createFromPixelTextureLayout(false);
+    }
+    return this.fromPixelTextureLayout;
+  }
+
+  private createFromPixelTextureLayout(useImport: boolean): WebGPULayout {
+    const bindGroupLayoutEntries: GPUBindGroupLayoutEntry[] = [];
+    // Output buffer binding layout.
+    bindGroupLayoutEntries.push({
+      binding: 0,
+      visibility: GPUShaderStage.COMPUTE,
+      buffer: {type: 'storage' as const}
+    });
+    // Input texture binding layout.
+    if (useImport) {
+      bindGroupLayoutEntries.push({
+        binding: 1,
+        visibility: GPUShaderStage.COMPUTE,
+        externalTexture: {},
+      });
+    } else {
+      bindGroupLayoutEntries.push(
+          {binding: 1, visibility: GPUShaderStage.COMPUTE, texture: {}});
+    }
+    // Uniform buffer binding layout.
+    bindGroupLayoutEntries.push(
+        {binding: 2, visibility: GPUShaderStage.COMPUTE, buffer: {}});
+    const fromPixelBindGroupLayout =
+        this.device.createBindGroupLayout({entries: bindGroupLayoutEntries});
+    const fromPixelPipelineLayout = this.device.createPipelineLayout(
+        {bindGroupLayouts: [fromPixelBindGroupLayout]});
+    return {
+      bindGroupLayout: fromPixelBindGroupLayout,
+      pipelineLayout: fromPixelPipelineLayout
+    };
+  }
+
+  private copyExternalImageToTexture(
+      externalImage: ExternalImage|HTMLVideoElement,
+      outShape: number[]): GPUTextureView {
+    const textureUsage = GPUTextureUsage.COPY_DST |
+        GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.TEXTURE_BINDING;
+    const textureFormat = 'rgba8unorm' as GPUTextureFormat;
+    const texture = this.textureManager.acquireTexture(
+        outShape[1], outShape[0], textureFormat, textureUsage);
+    const externalResource = texture.createView();
+
+    this.queue.copyExternalImageToTexture(
+        {source: externalImage as ExternalImage, origin: {x: 0, y: 0}},
```

Comment:
origin is default value here, don't need to explicit set them(I know it's my fault in previous code).

Reply:
Thanks, done